OpenRA.Mods.Common/Traits/BotModules/McvManagerBotModule.cs

```diff
@@ -197,8 +201,24 @@ void DeployMcv(IBot bot, Actor mcv, bool move)
 
 			var baseCenter = GetRandomBaseCenter();
 
-			return FindPos(baseCenter, baseCenter, Info.MinBaseRadius,
+			// Select a Spot in the base.
+			var selectedPosition = FindPos(baseCenter, baseCenter, Info.MinBaseRadius,
 				distanceToBaseIsImportant ? Info.MaxBaseRadius : world.Map.Grid.MaximumTileSearchRange);
+
+			// Otherwise select a location near resource fields.
+			if (resourceLayer != null && !distanceToBaseIsImportant)
```

Comment:
It should be moved in near furture to a more usable and suitable Location/file. Only an Resourcefield is ugly... It would be better when the bot can decide between TargetLocations.

Reply:
it would be best to have some sort of clustering algorithm so that the AI would deploy to resource fields, and not the remains of a harvester